packages/web/components/bridge/immersive/quote-detail.tsx

```diff
@@ -0,0 +1,244 @@
+import { isNil } from "@osmosis-labs/utils";
+import classNames from "classnames";
+import { FunctionComponent, PropsWithChildren } from "react";
+
+import { Icon } from "~/components/assets";
+import { Tooltip } from "~/components/tooltip";
+import { t } from "~/hooks";
+
+import { BridgeProviderDropdown } from "./bridge-provider-dropdown";
+import { BridgeQuoteRemainingTime } from "./bridge-quote-remaining-time";
+import { BridgeQuotes } from "./use-bridge-quotes";
+
+/** Bridge quotes result type, but with some modifications to require a quote to be selected. */
+export type BridgeQuotesDetails = Omit<BridgeQuotes, "selectedQuote"> & {
+  selectedQuote: NonNullable<BridgeQuotes["selectedQuote"]>;
+};
+
+/** Selected bridge provider, with dropdown to select a different one. */
+export const BridgeProviderDropdownRow: FunctionComponent<
+  BridgeQuotesDetails
+> = ({
+  selectedQuote,
+  successfulQuotes,
+  setSelectedBridgeProvider,
+  isRefetchingQuote,
+}) => (
+  <QuoteDetailRow label={t("transfer.provider")} isLoading={isRefetchingQuote}>
+    <BridgeProviderDropdown
+      selectedQuote={selectedQuote}
+      quotes={successfulQuotes}
+      onSelect={(bridgeId) => setSelectedBridgeProvider(bridgeId)}
+    />
+  </QuoteDetailRow>
+);
```

Comment:
I believe we should be cautious about passing all state values to a component. Any reference change can trigger a re-render, even if we use a memo function. In my opinion, we should only pass the necessary props to a component. This approach simplifies the code, making it easier to read and maintain while also reducing re-renders when required